lib/features/children/details/pages/child_details_page.dart

```diff
@@ -0,0 +1,95 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:givt_app/app/routes/routes.dart';
+import 'package:givt_app/core/enums/amplitude_events.dart';
+import 'package:givt_app/features/children/details/cubit/child_details_cubit.dart';
+import 'package:givt_app/features/children/details/widgets/child_details_item.dart';
+import 'package:givt_app/features/children/details/widgets/child_giving_allowance_card.dart';
+import 'package:givt_app/l10n/l10n.dart';
+import 'package:givt_app/utils/utils.dart';
+import 'package:go_router/go_router.dart';
+
+class ChildDetailsPage extends StatelessWidget {
+  const ChildDetailsPage({
+    super.key,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final locals = AppLocalizations.of(context);
+    return BlocBuilder<ChildDetailsCubit, ChildDetailsState>(
+      builder: (context, state) {
+        return Scaffold(
+          appBar: AppBar(
+            actions: [
+              BackButton(
+                onPressed: () {
+                  context.pop();
+                  AnalyticsHelper.logEvent(
+                    eventName: AmplitudeEvents.backClicked,
+                  );
+                },
+              ),
+              const Spacer(),
+              if (state is ChildDetailsFetchedState)
+                TextButton.icon(
+                  icon: const Icon(Icons.edit),
+                  label: Text(
+                    locals.budgetExternalGiftsEdit,
+                    style: Theme.of(context).textTheme.titleMedium?.copyWith(
+                          fontWeight: FontWeight.bold,
+                          color: AppTheme.inputFieldBorderSelected,
+                        ),
+                  ),
+                  onPressed: () {
+                    context.pushNamed(
+                      Pages.editChild.name,
+                      extra: context.read<ChildDetailsCubit>(),
+                    );
+                  },
+                ),
+            ],
+            automaticallyImplyLeading: false,
+            backgroundColor: state is ChildDetailsFetchedState
+                ? state.profileDetails.profile.monsterColor
+                : null,
+          ),
+          body: state is ChildDetailsFetchingState
```

Comment:
I've seen a lot of these kind of conditionals into the registration part as well. Coming back to it after a while it becomes confusing and creates some overhead. What if you would put here the builder and then based on the state you return the desired widget?

Reply:
hmm, questionable for me. I like this style in general